### ISSUE TITLE
Use a PVC to store Traefik LetsEncrypt Certificates?

### ISSUE THREAD

**Issue by MartiniMoe** [kube-hetzner/kube-hetzner#123]
Hi there,

on my learning cluster I have the problem that Traefik always requests new certificates from LetsEncrypt when something happens to the Traefik Pod. Because of that I sooner or later always end up in LetsEncrypt rate limiting.

Would it be a good idea to somehow have a persistent volume where Traefik can store certificates, so they it does not need to request new ones when the pod gets recreated or something?

**Comment by mysticaltech** [comment]
Hey @MartiniMoe, this is the way Traefik works, so unless you are doing actual testing, you should never hit the rate limits. But if you are indeed testing, just use the staging le issuing authority instead.

Add those to your traefik_config:

```
"--log.level=DEBUG"
"--certificatesresolvers.myresolver.acme.caserver=https://acme-staging-v02.api.letsencrypt.org/directory"
```

Now, definitely, the PVC thing is an option I have read and should be possible, but I do not know how exactly myself. I would just recommend using the staging env for testing, it seems like the way it's supposed to work.

**Comment by nupplaphil** [comment]
I'm using the cert-manager to create a LE certificate as an secret and traefik is capable of using these kind of certificates when annotating it at the ingress. So there's no binding to a node or pvc necessary :)

If you are wanting to continue to run Traefik Community Edition, LetsEncrypt HA can be achieved by using a Certificate Controller such as Cert-Manager. When using Cert-Manager to manage certificates, it will create secrets in your namespaces that can be referenced as TLS secrets in your ingress objects. 
Quote of https://doc.traefik.io/traefik/v2.0/providers/kubernetes-crd/

**Comment by mysticaltech** [comment]
Thanks for the tip @nupplaphil, awesome!!

**Comment by MartiniMoe** [comment]
Thanks for your input!

If it is possible to not use cert-manager I would like to only use Traefik. As @mysticaltech pointed out it should work since this is the way Traefik is supposed to work.

So maybe I have a different problem with my cluster and the certificates. My ingresses at first work and Traefik delivers LetsEncrypt certificates. A few days later it suddenly stops working. Traefik delivers its default certificate and I can find log entries like "Unable to obtain challenge data" or something like that. I will post logs later today.

**Comment by mysticaltech** [comment]
@MartiniMoe, @nupplaphil is right, the problem with using TLS with Traefik is that unless you go with the enterprise version, it's designed to be non-HA, your site will actually go down whenever it regenerates or whenever something happens. And cert-manager instead makes it so simple, it generates the certificate, which you can use inside Traefik as if it was a static certificate, while not having any downtime!

Basically, it's simpler and better. So if you wanted, just remove all TLS config from Traefik, and use cert-manager for that instead (you can install it with helm, it's really simple), while still using the certificates mentions in your ingress definitions. Most probably will work like a charm and is definitely a very elegant solution IMHO.

**Comment by MartiniMoe** [comment]
Thinks I will look into cert-manager then and close this issue :)
